router/batchrouter/asyncdestinationmanager/snowpipestreaming/snowpipestreaming.go

```diff
@@ -268,14 +279,18 @@ func (m *Manager) Upload(asyncDest *common.AsyncDestinationStruct) common.AsyncU
 	m.stats.jobs.importing.Count(len(importingJobIDs))
 	m.stats.jobs.failed.Count(len(failedJobIDs))
 
-	return common.AsyncUploadOutput{
+	asyncUploadOutput := common.AsyncUploadOutput{
 		ImportingJobIDs:     importingJobIDs,
 		ImportingCount:      len(importingJobIDs),
 		ImportingParameters: importParameters,
 		FailedJobIDs:        failedJobIDs,
 		FailedCount:         len(failedJobIDs),
 		DestinationID:       asyncDest.Destination.ID,
 	}
+	if failedReason != nil {
+		asyncUploadOutput.FailedReason = failedReason.Error()
+	}
+	return asyncUploadOutput
```

Comment:
Currently `failedReason` is not being returned in the output. Made changes to include it only for validations error. If required, we can add it for other cases as well.

Reply:
Yeah, let's include it for other cases as well. It's should be a very small change.

Reply:
Done. Now sending the very first error in case of multiple `uploadInfos`